File: app/src/main/java/com/semisonfire/cloudgallery/ui/dialogs/AlertDialogFragment.java
package com.semisonfire.cloudgallery.ui.dialogs;

import android.app.AlertDialog;
import android.app.Dialog;
import android.os.Bundle;
import android.support.annotation.NonNull;

import com.semisonfire.cloudgallery.ui.dialogs.base.BaseDialogFragment;

public class AlertDialogFragment extends BaseDialogFragment {

    private static final String ARGUMENT_TITLE = "ARGUMENT_TITLE";
    private static final String ARGUMENT_MESSAGE = "ARGUMENT_MESSAGE";
    private static final String ARGUMENT_COLOR = "ARGUMENT_COLOR";

    private String mTitle;
    private String mMessage;
    private int mButtonColor = -1;

    public static AlertDialogFragment newInstance(String title, String message, int color) {
        Bundle args = new Bundle();
        args.putString(ARGUMENT_TITLE, title);
        args.putString(ARGUMENT_MESSAGE, message);
        args.putInt(ARGUMENT_COLOR, color);

        AlertDialogFragment alertDialogFragment = new AlertDialogFragment();
        alertDialogFragment.setArguments(args);
        return alertDialogFragment;
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {

        Bundle args = getArguments();
        if (args != null) {
            mTitle = args.getString(ARGUMENT_TITLE);
            mMessage = args.getString(ARGUMENT_MESSAGE);
            mButtonColor = args.getInt(ARGUMENT_COLOR, -1);
        }

        AlertDialog dialog = new AlertDialog.Builder(getContext())
                .setTitle(mTitle != null ? mTitle : "")
                .setMessage(mMessage != null ? mMessage : "")
                .setPositiveButton(android.R.string.ok, (dialogInterface, i) -> {
                    if (getDialogListener() != null) {
                        getDialogListener().onPositiveClick(dialogInterface);
                    }
                })
                .setNegativeButton(android.R.string.cancel, (dialogInterface, i) -> {
                    if (getDialogListener() != null) {
                        getDialogListener().onNegativeClick(dialogInterface);
                    }
                })
                .create();
        dialog.setOnShowListener(dialogInterface -> {
            if (mButtonColor != -1) {
                dialog.getButton(AlertDialog.BUTTON_NEGATIVE).setTextColor(mButtonColor);
                dialog.getButton(AlertDialog.BUTTON_POSITIVE).setTextColor(mButtonColor);
            }
        });
        return dialog;
    }
}

File: app/src/main/java/com/semisonfire/cloudgallery/ui/dialogs/AlertDialogFragment.kt
package com.semisonfire.cloudgallery.ui.dialogs

import android.app.AlertDialog
import android.app.Dialog
import android.os.Bundle
import androidx.fragment.app.DialogFragment
import com.semisonfire.cloudgallery.R

class AlertDialogFragment : DialogFragment() {

    var dialogListener: DialogListener? = null

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        var dialogTitle = ""
        var dialogMessage = ""
        var dialogButtonColor = -1

        arguments?.apply {
            dialogTitle = getString(ARGUMENT_TITLE) ?: ""
            dialogMessage = getString(ARGUMENT_MESSAGE) ?: ""
            dialogButtonColor = getInt(ARGUMENT_COLOR, -1)
        }

        val dialog = AlertDialog.Builder(context)
            .setTitle(dialogTitle)
            .setMessage(dialogMessage)
            .setPositiveButton(R.string.action_ok) { dialogInterface, i ->
                dialogListener?.onPositiveClick()
                dialogInterface.cancel()
            }
            .setNegativeButton(R.string.action_cancel) { dialogInterface, i ->
                dialogListener?.onNegativeClick()
                dialogInterface.cancel()
            }
            .create()
        dialog.setOnShowListener {
            if (dialogButtonColor != -1) {
                dialog.getButton(AlertDialog.BUTTON_NEGATIVE).setTextColor(dialogButtonColor)
                dialog.getButton(AlertDialog.BUTTON_POSITIVE).setTextColor(dialogButtonColor)
            }
        }
        return dialog
    }

    companion object {

        private const val ARGUMENT_TITLE = "ARGUMENT_TITLE"
        private const val ARGUMENT_MESSAGE = "ARGUMENT_MESSAGE"
        private const val ARGUMENT_COLOR = "ARGUMENT_COLOR"

        @JvmStatic
        fun newInstance(
            title: String?,
            message: String?,
            color: Int
        ): AlertDialogFragment {

            val args = Bundle()
            args.putString(ARGUMENT_TITLE, title)
            args.putString(ARGUMENT_MESSAGE, message)
            args.putInt(ARGUMENT_COLOR, color)

            val alertDialogFragment = AlertDialogFragment()
            alertDialogFragment.arguments = args
            return alertDialogFragment
        }
    }
}